### PR TITLE
automake: implement make check and make installcheck

### PR DESCRIPTION
**Pull request by j-ogas**
Addresses: #603.

Note that when `make check` executes all the output is piped into a `test-suite.log` file leaving the terminal inactive for several minutes before reporting success.

### CODE REVIEW

File: bin/ch-test [hunk 1]
```diff
@@ -840,10 +839,22 @@ if [[ $phase = one-file ]]; then
...
+if [[ -z $phase && -n $CH_TEST_PHASE ]]; then
+    case $CH_TEST_PHASE in
+    all|build|examples|run)
+        phase=$CH_TEST_PHASE
+    ;;
+    *)
+    ;;
+    esac
+fi
+
```
Comment (reidpr) [inline bin/ch-test:852]: I think we should just copy `$CH_TEST_PHASE` to `$phase` without checking it. That will be sufficient for internal purposes, and at some point in the future we can write code that checks for valid phase just once.

File: test/build/10_sanity.bats [hunk 2]
```diff
@@ -50,10 +50,20 @@ load ../common
             ch-checkns)
                 ;;
             *)
+                # CH_TEST_PHASE is an undocumented variable that allows us to
+                # execute ch-test with the automake without arguments.
+                local phase
+                if [[ -n $CH_TEST_PHASE ]]; then
+                    phase=$CH_TEST_PHASE
+                    unset CH_TEST_PHASE
+                fi
```
Comment (reidpr) [inline test/build/10_sanity.bats:59]: It's unclear to me what adding this tests?

File: .gitignore [hunk 3]
```diff
@@ -60,10 +60,13 @@ Makefile.in
 /aclocal.m4
 /autom4te.cache/
 /bin/.deps/
+/bin/ch-test.log
+/bin/ch-test.trs
```
Comment (reidpr) [inline .gitignore:64]: These will need to be added to `autogen.sh` too (see comment).

File: test/build/10_sanity.bats (hunk 2 above)
Comment (j-ogas) [reply test/build/10_sanity.bats:59]: This prevents the `executables sane` test from failing and bats from executing `ch-test all`. The documented and expected behavior for `./bin/ch-*` executables is print usage and exit. Without the above, `ch-test` will execute with the phase specified in `CH_TEST_PHASE` when executed with `make check`.

Comment (reidpr) [reply test/build/10_sanity.bats:59]: Aha, OK. Would it work to simply `unset CH_TEST_PHASE`?

**Comment by j-ogas** [comment]
It seems we are unable to redirect the output generated by `make check` to `stdout` instead of `test-suite.log`. This lack of terminal activity makes our travis tests time out.

Marking as differed and closing.

**Pull request closed without merging**